pinot-core/src/main/java/com/linkedin/pinot/core/data/manager/realtime/LLRealtimeSegmentDataManager.java

```diff
@@ -525,13 +531,12 @@ private boolean buildSegmentInternal(boolean forCommit) {
     // Build a segment from in-memory rows.If buildTgz is true, then build the tar.gz file as well
     // TODO Use an auto-closeable object to delete temp resources.
     File tempSegmentFolder = new File(_resourceTmpDir, "tmp-" + _segmentNameStr + "-" + String.valueOf(now()));
-
+    int nPartitions = _consumerWrapper.getPartitionCount(_kafkaTopic, /*maxWaitTimeMs=*/5000L);
```

Comment:
To add, is this additional call actually required? I thought LLC segment names have partition number encoded in the name, which would imply that we already queried the partition once. Can we not use the same value? Or does that happen in the controller?